src/appleseed/foundation/math/aabb.h

```diff
@@ -191,31 +164,101 @@ template <typename T, size_t N> AABB<T, N>& operator*=(AABB<T, N>& lhs, const T
 // Compute the surface area of a 3D bounding box.
 template <typename T> T half_surface_area(const AABB<T, 3>& bbox);
 template <typename T> T surface_area(const AABB<T, 3>& bbox);
+//
+// Floating N-dimensional axis-aligned bounding box [min, max] class.
+//
+
+template <typename T, size_t N>
+class FloatingAABB
+  : public AABB<T, N>
+{
+  public:
+    using AABB<T, N>::AABB;
```

Comment:
What's this?

Reply:
To inherit constructors

Reply:
Unlikely to work on VS 2012. I'll test.

Reply:
Thanks

Reply:
Seems like it's supported, I didn't know, cool!